### PR TITLE
[Supabase] New Triggers

### LINKED ISSUE #6537: [TRIGGER] Supabase DB Trigger or Realtime Notification
**User request:**
I have setup a simple workflow that needs to send an email based on records retrieved from a table in Supabase. This works fine when manually testing it… but this workflow is missing start trigger in Pipedream.

I would like workflow to be triggered as a result of a new record being added to this Supabase table… how do I configure Workflow to do that?

Supabase supports both Postgres DB Triggers and Real-time notifications; latter I think requires use of SUpabase JS SDK.

**Describe the event source. What app is this for, and what event does the trigger correspond to?**
Trigger when a new row is added to a table:

1. Polling trigger
2. Database Webhooks trigger + documentation/instructions on the README/source description on how to create the webhook through the Supabase UI (see below)

**Please provide a link to the relevant API docs for the specific service / operation this trigger is tied to.**
[Database Webhooks Docs](https://supabase.com/docs/guides/database/webhooks)
From my understanding of the docs, the only way to create a Database Webhook is through the Supabase UI (please verify if this is true)

**Context:**
[User request on Discourse](https://pipedream.com/community/t/how-do-i-use-a-supabase-db-trigger-or-supabase-realtime-notification-as-pipedream-trigger/6339/1)

### PR DESCRIPTION
**Pull request by andrewjschuang**
Resolves #6537.

### <samp>🤖 Generated by Copilot at c7d456c</samp>

This pull request adds two new sources (`new-row-added` and `new-webhook-event`) and updates the `select-row` action for the Supabase component. It also refactors some common props and constants, and adds more documentation for webhooks. These changes enhance the functionality and usability of the Supabase component.

### <samp>🤖 Generated by Copilot at c7d456c</samp>

> _Supabase component_
> _Webhooks, props, and sources_
> _`README` for spring_

## WHY

- Add a new polling-based trigger for new rows in a table.
- Add a webhook-based events trigger, but needs user configuration in the Supabase Web UI.

### <samp>🤖 Generated by Copilot at c7d456c</samp>

*  Add two new sources to the Supabase component: `new-row-added` and `new-webhook-event` ([link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-a17220b7f1068c80704b2c6ea7cde76d835a3e355dcb44c55bfe5ab8740a3f15R1-R110), [link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-84da0e60472ebe33479f4a5adde9e812297821021fe665cde3e25a2c8960b06fR1-R41))
  * `new-row-added` emits an event for every new row added in a table, using the `timer` interface and the `db` service ([link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-a17220b7f1068c80704b2c6ea7cde76d835a3e355dcb44c55bfe5ab8740a3f15R1-R110))
  * `new-webhook-event` emits an event for every `insert`, `update`, or `delete` operation in a table, using the `http` interface and the `deploy` hook ([link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-84da0e60472ebe33479f4a5adde9e812297821021fe665cde3e25a2c8960b06fR1-R41))
  * Both sources use the common props from the `base` object, which exports the `supabase` app, the `table` name, and the `rowIdentifier` column ([link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-cb35ea5e041e47ab56d54c0822ea2eee24715cb64f8f7ec80a36c3de11889cb1R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-a17220b7f1068c80704b2c6ea7cde76d835a3e355dcb44c55bfe5ab8740a3f15R1-R110), [link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-84da0e60472ebe33479f4a5adde9e812297821021fe665cde3e25a2c8960b06fR1-R41))
* Simplify the `sortOrder` prop of the `select-row` action by using the `propDefinition` property to reference the common definition from the `supabase` app file ([link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-a3a99b5e91cf4f5da0a5afc5937c42711db4e1aa81d8e17f1931825e04462d9eL48-R51), [link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-c8ec263b188fccb48a601a6d630ddce8a71dd4e3c00997cb80e0f241536f1021R34-R44))
* Add two new constants to the `constants` file: `MAX_OFFSET` and `HISTORICAL_EVENT_LIMIT`, which are used to limit the number of rows fetched from the Supabase database in the `new-row-added` and `new-webhook-event` sources, respectively ([link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-a51215ea811fdd2341cfa8eaa7d21ad6a5ea1f63b9a27fd129861ede78df1c69R38-R39), [link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-a17220b7f1068c80704b2c6ea7cde76d835a3e355dcb44c55bfe5ab8740a3f15R1-R110), [link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-84da0e60472ebe33479f4a5adde9e812297821021fe665cde3e25a2c8960b06fR1-R41))
* Add a new section to the README file, explaining how to configure and use webhooks with Supabase and Pipedream ([link](https://github.com/PipedreamHQ/pipedream/pull/6546/files?diff=unified&w=0#diff-bbdd382549c10a1297b4a5c3887d5cb0948557175fb54f398c8f8a6c4bd85e19R23-R34))

### CODE REVIEW

File: components/supabase/actions/select-row/select-row.mjs [hunk 1]
```diff
@@ -5,7 +5,7 @@ export default {
   key: "supabase-select-row",
   name: "Select Row",
   type: "action",
-  version: "0.0.2",
+  version: "0.0.3",
   description: "Selects row(s) in a database. [See the docs here](https://supabase.com/docs/reference/javascript/select)",
   props: {
     supabase,
```
Comment (dylburger) [inline components/supabase/actions/select-row/select-row.mjs:11]:
```suggestion
  description: "Select a row from a table. [See the documentation here](https://supabase.com/docs/reference/javascript/select)",
```

Comment (dylburger) [reply components/supabase/actions/select-row/select-row.mjs:11]: Changed "See documentation here" to "See the documentation here" to follow Google's Developer docs style guide.

File: components/supabase/supabase.app.mjs [hunk 2]
```diff
@@ -31,6 +31,17 @@ export default {
       description: "Filter type for the query",
       options: constants.FILTER_OPTIONS,
     },
+    sortOrder: {
+      type: "string",
+      label: "Sort Order",
+      description: "Sort ascending or descending",
```
Comment (dylburger) [inline components/supabase/supabase.app.mjs:37]:
```suggestion
      description: "Sort ascending or descending.",
```

Comment (dylburger) [reply components/supabase/supabase.app.mjs:37]: Added a period at the end of the description to follow Google's Developer docs style guide.

File: components/supabase/sources/new-row-added/new-row-added.mjs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import {
+  DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+  ConfigurationError,
+} from "@pipedream/platform";
+import constants from "../../common/constants.mjs";
+
+export default {
+  key: "supabase-new-row-added",
+  name: "New Row Added",
+  description: "Emit new event for every new row added in a table. [See documentation here](https://supabase.com/docs/reference/javascript/select)",
```
Comment (dylburger) [inline components/supabase/sources/new-row-added/new-row-added.mjs:11]:
```suggestion
  description: "Emit a new event for every new row added in a table. [See the documentation here](https://supabase.com/docs/reference/javascript/select)",
```

File: components/supabase/sources/new-row-added/new-row-added.mjs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        "column",
+      ],
+      optional: true,
+    },
+    filter: {
+      propDefinition: [
+        base.props.supabase,
+        "filter",
+      ],
+      optional: true,
```
Comment (dylburger) [inline components/supabase/sources/new-row-added/new-row-added.mjs:28]:
```suggestion
      description: "The name of the table to watch for new rows.",
```

Comment (dylburger) [reply components/supabase/sources/new-row-added/new-row-added.mjs:28]: Added a period at the end of the description to follow Google's Developer docs style guide.

File: components/supabase/sources/new-row-added/new-row-added.mjs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        "sortOrder",
+      ],
+    },
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
```
Comment (dylburger) [inline components/supabase/sources/new-row-added/new-row-added.mjs:57]:
```suggestion
      description: "Column name to order by.",
```

File: components/supabase/sources/new-row-added/new-row-added.mjs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  methods: {
+    _getOffset() {
+      return this.db.get("offset") || 0;
+    },
+    _setOffset(offset) {
+      this.db.set("offset", offset);
+    },
+  },
+  async run() {
+    const {
```
Comment (dylburger) [inline components/supabase/sources/new-row-added/new-row-added.mjs:68]:
```suggestion
      description: "The column name to use as the row identifier.",
```

Comment (dylburger) [reply components/supabase/sources/new-row-added/new-row-added.mjs:68]: Added a period at the end of the description to follow Google's Developer docs style guide.

File: components/supabase/sources/new-webhook-event/new-webhook-event.mjs [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import base from "../common/base.mjs";
+import constants from "../../common/constants.mjs";
+
+export default {
+  key: "supabase-new-webhook-event",
+  name: "New Webhook Event (Instant)",
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/supabase/sources/new-webhook-event/new-webhook-event.mjs:6]: Changed "Emit new event" to "Emit a new event" and "See documentation here" to "See the documentation here" to follow Google's Developer docs style guide.

File: components/supabase/sources/new-webhook-event/new-webhook-event.mjs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  props: {
+    ...base.props,
+    http: {
+      type: "$.interface.http",
+    },
+  },
+  hooks: {
+    async deploy() {
+      const client = await this.supabase._client();
+      const query = client
```
Comment (dylburger) [inline components/supabase/sources/new-webhook-event/new-webhook-event.mjs:19]:
```suggestion
      description: "The name of the table to watch for new rows.",
```

Comment (dylburger) [reply components/supabase/sources/new-webhook-event/new-webhook-event.mjs:19]: Added a period at the end of the description to follow Google's Developer docs style guide.

File: components/supabase/sources/new-webhook-event/new-webhook-event.mjs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        .from(this.table)
+        .select()
+        .range(0, constants.HISTORICAL_EVENT_LIMIT);
+      const { data } = await query;
+      for (const row of data) {
```
Comment (dylburger) [inline components/supabase/sources/new-webhook-event/new-webhook-event.mjs:24]:
```suggestion
      description: "The column name to use as the row identifier.",
```

Comment (dylburger) [reply components/supabase/sources/new-webhook-event/new-webhook-event.mjs:24]: Added a period at the end of the description to follow Google's Developer docs style guide.

**Review by luancazarine — APPROVED** [review]
Hi @andrewjschuang, thanks for your contribution! LGTM!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Supabase_New_Triggers_6546.pdf](https://github.com/PipedreamHQ/pipedream/files/11526374/Supabase_New_Triggers_6546.pdf)

**Pull request merged**